### PR TITLE
Feature/observe names

### PR DESCRIPTION
**Pull request by wangyoucao577**
# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.
Closes https://github.com/Telenav/osrm-backend/issues/123

Please read our [documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/releasing.md) on release and version management.
If your PR is still work in progress please attach the relevant label.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?
doc: https://github.com/Telenav/open-source-spec/blob/master/osrm/doc/osrm-toolchain-files/map.osrm.names.md

### CODE REVIEW

File: integration/osrmfiles/tar_loader.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func Load(contents ContentsOperator) error {
 			break // End of archive
 		}
 		if err != nil {
-			glog.Fatal(err)
+			glog.Errorf("parsing tar file %s failed, err: %v", contents.FilePath(), err)
+			break
```
Comment (CodeBear801) [inline integration/osrmfiles/tar_loader.go]: Who will handle partial failure, when part of data be load into writer?

File: integration/osrmfiles/osrmtype/nametable/sum2bits.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package nametable
+
+// sum2bits calculates summation of 16 2-bit values using SWAR
+// https://github.com/Telenav/open-source-spec/blob/master/osrm/doc/sum2bits-swar.md
+// C++ implementation: https://github.com/Telenav/osrm-backend/blob/b24b8a085dc10bea279ffb352049330beae23791/include/util/indexed_data.hpp#L94
+func sum2Bits(value uint32) uint32 {
+	value = (value >> 2 & 0x33333333) + (value & 0x33333333)
+	value = (value >> 4 & 0x0f0f0f0f) + (value & 0x0f0f0f0f)
+	value = (value >> 8 & 0x00ff00ff) + (value & 0x00ff00ff)
+	return (value >> 16 & 0x0000ffff) + (value & 0x0000ffff)
```
Comment (CodeBear801) [inline integration/osrmfiles/osrmtype/nametable/sum2bits.go]: Your document is very helpful, need time to digest this part from OSRM.

File: integration/osrmfiles/tar_loader.go (hunk 1 above)
Comment (wangyoucao577) [reply integration/osrmfiles/tar_loader.go]: Yes. I have met issue that a file broked but some of the contents still possible to load, so just print a log and go with correct parts

File: integration/osrmfiles/osrmtype/nametable/indexed_data.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	return i.values[dataStart:dataEnd], nil
+}
+
+// C++ implementation: https://github.com/Telenav/osrm-backend/blob/b24b8a085dc10bea279ffb352049330beae23791/include/util/indexed_data.hpp#L70
+func (i *IndexedData) varAdvance(byteLen, dataStart, lengthStart uint32) (uint32, uint32) {
+	if byteLen > twoBitsMax { // byteLen should only in 2-bits
+		glog.Fatalf("invalid byteLen %d", byteLen)
+	}
+
```
Comment (CodeBear801) [inline integration/osrmfiles/osrmtype/nametable/indexed_data.go]: May be a comment is helpful here:
// Nothing to do for varAdvance while byteLen == 0

Comment (wangyoucao577) [reply integration/osrmfiles/osrmtype/nametable/indexed_data.go]: Done.

**Pull request merged**